test/integration/push-jenkins-update.test.js

```diff
@@ -14,74 +14,93 @@ require('../../scripts/jenkins-status')(app, events)
 tap.test('Sends POST requests to https://api.github.com/repos/nodejs/node/statuses/<SHA>', (t) => {
   const jenkinsPayload = readFixture('success-payload.json')
 
-  const prCommitsScope = setupGetCommitsMock('node')
-  const scope = nock('https://api.github.com')
+  setupGetCommitsMock('node')
+    .on('replied', (req, interceptor) => {
+      t.doesNotThrow(() => interceptor.scope.done())
+    })
+  nock('https://api.github.com')
     .filteringPath(ignoreQueryParams)
     .post('/repos/nodejs/node/statuses/8a5fec2a6bade91e544a30314d7cf21f8a200de1')
     .reply(201)
+    .on('replied', (req, interceptor) => {
+      t.doesNotThrow(() => interceptor.scope.done())
+    })
 
-  t.plan(1)
+  t.plan(3)
 
   supertest(app)
     .post('/node/jenkins/start')
     .send(jenkinsPayload)
-    .expect(201)
+    .expect(200)
     .end((err, res) => {
-      prCommitsScope.done()
-      scope.done()
       t.equal(err, null)
     })
 })
 
 tap.test('Allows repository name to be provided with URL parameter when pushing job started', (t) => {
   const jenkinsPayload = readFixture('pending-payload.json')
 
-  const prCommitsScope = setupGetCommitsMock('citgm')
-  const scope = nock('https://api.github.com')
+  setupGetCommitsMock('citgm')
+    .on('replied', (req, interceptor) => {
+      t.doesNotThrow(() => interceptor.scope.done())
+    })
+  nock('https://api.github.com')
     .filteringPath(ignoreQueryParams)
     .post('/repos/nodejs/citgm/statuses/8a5fec2a6bade91e544a30314d7cf21f8a200de1')
     .reply(201)
+    .on('replied', (req, interceptor) => {
+      t.doesNotThrow(() => interceptor.scope.done())
+    })
```

Comment:
Interesting alternative approach to verifying scopes that I haven't used before. Any obvious upsides compared to the previous approach?

Reply:
Now that we're using EventEmitter and Promises, the timing is different and as a result the callback below to run before we made a call to the `scope` URL, so we get an error when we call `scope.done()`

Reply:
Ahh okey, so that's another approach to fixing what was my biggest headache in https://github.com/nodejs/github-bot/pull/258 basically.

What I ended up doing there, was to make sure all event handlers returned a `Promise`. That way the resulting `Promise` coming out of the `app.emitJenkinsEvent()`-equivalent of https://github.com/nodejs/github-bot/pull/258 would actually resolve after all handlers had completed. In the end we can be sure that all handlers (and thereby `scope`) has finished when the request gets the 200 response back.

No biggie tho, just wanted to understand the pros & reasons behind this new approach introduced.

Reply:
That works too, but no reason to delay the response until app.emitJenkinsEvent finishes since Jenkins doesn't do anything with 200/500 status codes. We could for testability if we wanted to, but this approach seems fine and more robust?